tools/src/scie_pants/pants_version.py

```diff
@@ -148,9 +148,16 @@ def determine_latest_stable_version(
     ptex: Ptex, pants_config: Path, find_links_dir: Path, github_api_bearer_token: str | None = None
 ) -> tuple[Callable[[], None], ResolveInfo]:
     info(f"Fetching latest stable Pants version since none is configured")
-    pants_version = ptex.fetch_json("https://pypi.org/pypi/pantsbuild.pants/json")["info"][
-        "version"
-    ]
+
+    latest_tag = ptex.fetch_json(
+        "https://github.com/pantsbuild/pants/releases/latest", Accept="application/json"
+    )["tag_name"]
+    if not latest_tag.startswith("release_"):
+        fatal(
+            f'Expected the GitHub Release tagged "latest" to have the "release_" prefix. Got "{latest_tag}"'
+        )
+
+    pants_version = latest_tag[len("release_") :]
```

Comment:
Like to avoid this kind of prefix checking when possible. How about:
```suggestion
    prefix, _, pants_version = latest_tag.partition("_")
    if prefix != "release" or not pants_version:
        fatal(
            f'Expected the GitHub Release tagged "latest" to have the "release_" prefix. Got "{latest_tag}"'
        )
```

Reply:
Idk. I think the prefix checking is a little repetitive, but is closer to the intent we're trying to translate into code.

Quite literally, what I want to represent (in English) is "is this tag doesn't start with 'release_' something is wrong, otherwise the version is the rest of the tag"

And this code _directly_ translates to that (which usually code doesn't get to directly translate from intent)

If we had Python 3.9, we could probably use `stripprefix`, but we don't.

Reply:
Also your code assumes the tag will have 3 parts. But the whole point of checking is that it may not be what we expect. So that'll result in a TypeError (or whatever the unpack equivalent is)

Reply:
no, my code has no assumptions besides it being a string. `str.partition()` always returns a 3-tuple. If there are multiple instances of the partitioning charcter, the first will be the split point and the the rest preserved in the third element (the version, in this case).

My code reads, literally: "Split this string into two at the first occurrence of `_`, if the first part is "release" and we have a second part, that is the version, fail otherwise" (which basically is the same as yours using more words, but closer to what the code actually does.)

I like this more than what you have based on: fewer lines (while still being readable), not repeating the `"release_"` prefix, `len("release_")` is icky (but better than `8`.)

Reply:
I'm convinced. :+1: